### PR TITLE
Fix EZP-24521: Performance in ezxmltexttype attribute deletion

### PR DESCRIPTION
**Pull request by joaoinacio**
JIRA: https://jira.ez.no/browse/EZP-24521

In eZContentObject::purge(),  for every attribute attribute (for every version) the deleteStoredObjectAttribute() function is called.

For contents with multiple versions, removing the urls linked to ezxml contents can have an impact on performance.

This optimizes `eZXMLTextType::deleteStoredObjectAttribute()` by executing this operation only once per attribute when no version is given (removing data for all versions).

### CODE REVIEW

File: kernel/classes/datatypes/ezxmltext/ezxmltexttype.php [hunk 1]
```diff
@@ -818,6 +822,12 @@ function deleteStoredObjectAttribute( $contentObjectAttribute, $version = null )
 
             $db->query( "DELETE FROM ezurl WHERE id IN ($unusedUrlIDString)" );
         }
+
+        /* If all the versions/urls of the attribute where removed, do not try to remove them again */
```
Comment (joaoinacio) [inline kernel/classes/datatypes/ezxmltext/ezxmltexttype.php]: Nitpick: typo (were removed)

File: kernel/classes/datatypes/ezxmltext/ezxmltexttype.php [hunk 2]
```diff
@@ -843,6 +853,8 @@ function batchInitializeObjectAttributeData( $classAttribute )
         $xmlText = "'" . $db->escapeString( $xmlText ) . "'";
         return array( 'data_text' => $xmlText );
     }
+
+    protected $deletedStoredObjectAttribute;
```
Comment (andrerom) [inline kernel/classes/datatypes/ezxmltext/ezxmltexttype.php]: phpdoc: *List of fully deleted object attributes by id, used to know when we don't need to perform additional url cleanup*

**Comment by andrerom** [comment]
+1

**Comment by yannickroger** [comment]
+1

**Pull request merged**